### ISSUE TITLE
Add homebrew support for macOS

### ISSUE THREAD

**Issue by tzhao11** [trojan-gfw/trojan#63]
This will simplify the process of deploying trojan on macOS.

Formulas of shadowsocks and v2ray may be used as references:
https://github.com/v2ray/homebrew-v2ray/blob/master/Formula/v2ray-core.rb
https://github.com/Homebrew/homebrew-core/blob/master/Formula/shadowsocks-libev.rb

**Comment by GreaterFire** [comment]
Yes. I would like to package for homebrew given the user base right now. But they have very [strict rules](https://docs.brew.sh/Acceptable-Formulae) regarding whether a formula can be added to core:

> The software in question must:
> 
> be maintained (i.e. the last release wasn’t ages ago, it works without patching on all supported macOS releases and has no outstanding, unpatched security vulnerabilities)
> be known
> be stable (e.g. not declared “unstable” or “beta” by upstream)
> be used
> have a homepage
> We will reject formulae that seem too obscure, partly because they won’t get maintained and partly because we have to draw the line somewhere.
> 
> We frown on authors submitting their own work unless it is very popular.

Since they discourage authors to submit their software directly, we might need someone else to fork and commit and make the pull request.

**Comment by tzhao11** [comment]
Maybe creating a [tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap) first to test stuff, then add to core when it's mature enough?

**Comment by GreaterFire** [comment]
@tzhao11 Yes. I would do that in a few days.

**Comment by GreaterFire** [comment]
@tzhao11 Try the [tap](https://github.com/trojan-gfw/homebrew-trojan) via `brew install trojan-gfw/trojan/trojan`.

**Comment by tzhao11** [comment]
@GreaterFire Thanks, it works well. Would you consider change the default config to a client one since that's probably what most people using macOS would do?

**Comment by GreaterFire** [comment]
@tzhao11 Good idea.

**Comment by GreaterFire** [comment]
@tzhao11 Would you like to fork homebrew and make the pull request?
